integration_tests/test_remote.py

```diff
@@ -0,0 +1,65 @@
+from git_wrapper.repo import GitRepo
+
+
+def test_prune_tags(repo_root):
+    repo = GitRepo(repo_root)
+
+    repo.remote.add("test", f"file://{repo_root}")
+    remotes = repo.remote.names()
+    assert "test" in remotes
+
+    repo.tag.create("test_tag", "master")
+    assert "test_tag" in repo.tag.names()
```

Comment:
This is great btw, checking that we have the data we need where it should first before acting on it is a great starting point.